src/DotNetBoilerplate.Infrastructure/DAL/Handlers/Events/GetEventByIdHandler.cs

```diff
@@ -0,0 +1,24 @@
+﻿using DotNetBoilerplate.Application.Events;
+using DotNetBoilerplate.Application.Events.Responses;
+using DotNetBoilerplate.Core.Events;
+using DotNetBoilerplate.Infrastructure.DAL.Contexts;
+using DotNetBoilerplate.Shared.Abstractions.Queries;
+using Microsoft.EntityFrameworkCore;
+
+namespace DotNetBoilerplate.Infrastructure.DAL.Handlers.Events;
+
+internal sealed class GetEventByIdHandler(
+    DotNetBoilerplateReadDbContext dbContext
+    )
+: IQueryHandler<GetEventByIdQuery, EventInfoResponse>
+{
+    public async Task<EventInfoResponse> HandleAsync(GetEventByIdQuery query)
+    {
+        var @event = await dbContext.Events
+            .AsNoTracking()
+            .FirstOrDefaultAsync(e => e.Id == query.EventId);
+        
+        //If not null
+        return new EventInfoResponse {Title = @event.Title, Description = @event.Description};
```

Comment:
Dobrze kminisz, po prostu we wszystkich generykach obsłużmy sobie że response może być nullem, a dopiero w samym endpoincie określimy sobie co z tym zrobić np. rzucic 404 (to co pisałem w tym endponcie)